compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -323,22 +330,48 @@ static void addGPUBufferizePasses(OpPassManager &funcPassManager) {
   funcPassManager.addPass(createCSEPass());
 }
 
+/// Control function for decomposing pack and unpack ops. Returns true if the
+/// op is a PackOp with a DispatchTensorLoadOp producer, or an UnPackOp with
+/// only DispatchTensorStoreOp consumers.
+LogicalResult isAtBoundary(Operation *op) {
```

Comment:
Noting from offline conversation. We should use call backs this way. This is bad practice. We should instead make this behavior part of the pass. I think this is going to be a follow up.

Reply:
Having a control function this was is kind of giving me a pause. Can we avoid that?

Reply:
I removed the control function from the public functions and added a wrapper pass for decomposition at function boundaries.